src/client/Microsoft.Identity.Client/Utils/Guard.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+
+namespace Microsoft.Identity.Client.Utils
+{
+    static class Guard
+    {
+        public static T AgainstNull<T>(
+            [NotNull] T? argument,
+            [CallerArgumentExpression("argument")] string? paramName = null)
+            where T : class
+        {
+            if (argument is null)
+            {
+                throw new ArgumentNullException(paramName);
+            }
+
+            return argument;
+        }
+
+        public static string AgainstNullOrEmpty(
+            [NotNull] string? argument,
+            [CallerArgumentExpression("argument")] string? paramName = null)
+        {
+            if (string.IsNullOrEmpty(argument))
+            {
+                throw new ArgumentNullException(paramName);
+            }
+
+            return argument;
+        }
+        public static string AgainstNullOrWhitespace(
+            [NotNull] string? argument,
+            [CallerArgumentExpression("argument")] string? paramName = null)
+        {
+            if (string.IsNullOrWhiteSpace(argument))
+            {
+                throw new ArgumentNullException(paramName);
+            }
+
+            return argument;
+        }
+    }
+}
+
+#if NETFRAMEWORK || NETSTANDARD || NETCOREAPP2X
```

Comment:
not sure if NETCOREAPP2X is relevant? MSAL is compiled against NETSTANDARD and NET6

Reply:
sorry. i copied that from one of my projects. will fix